Java/syntax_test_java.java

```diff
@@ -687,7 +687,7 @@ public void someMethod(WithParam foo) throws Exception {
 //                                  ^ constant.numeric.java
 
     foo.forEach((k, v) -> {
-//                     ^ storage.type.lambda.java
+//                     ^ storage.type.function.lambda.java
```

Comment:
`lambda` seems to be the more general term and isn't language specific. I favor going for that instead of "arrow", which relates to the syntax more than the actual construct.